helper/schema/field_reader_config.go

```diff
@@ -228,7 +228,7 @@ func (r *ConfigFieldReader) readMap(k string, schema *Schema) (FieldReadResult,
 		return FieldReadResult{}, nil
 	}
 
-	var value interface{}
+	var value map[string]interface{}
```

Comment:
I'm confused as to why this works, since `Value` is typed as `interface{}` on `FieldReadResult`

Reply:
This is a common mistake in Go, since assigning a nil map to value create a non-nil interface: https://play.golang.org/p/wh9nf7wS4QR 

And conversely, if nothing is assigned (the computed case), then then a result that exists is expected to be a map, for example:
https://github.com/hashicorp/terraform/blob/master/helper/schema/field_reader_diff.go#L98

Reply:
Interesting, thanks!